### ISSUE TITLE
Create New Model From HasOne Relation Defined With 'withDefault'

### ISSUE THREAD

**Issue by MannikJ** [laravel/nova-issues#2532]
- Laravel Version: 6.18.8
- Nova Version: 2.12.0
- PHP Version: 7.4.5
- Database Driver & Version: MySql 8.0
- Operating System and Version: #Linux Mint 19.3
- Browser type and version: Chrome 81.0.4044.92

### Description:
I just encountered a problem when trying to create a new model instance via a hasOne relation defined with `withDefault()`. 

Nova displays a strange error notification: 

> The HasOne relation has already filled.

### Steps To Reproduce:
Create two tables with a hasOne relation and define the relation with `withDefault()` like so:

```php
    public function relation()
    {
        return $this->hasOne(Related::class, 'self_id')
            ->withDefault();
    }
```

**Comment by dnevnick** [comment]
Hello. Any updates on this issue?
Got the same problem, this is the only mention of this problem that I could find.

**Comment by crynobone** [comment]
@dnevnick thank you for highlighting this issue. I will look into it ASAP

**Comment by crynobone** [comment]
Fixed and released on v3.21.1
